router/h2/src/main/scala/com/twitter/finagle/buoyant/h2/netty4/Netty4ClientStreamTransport.scala

```diff
@@ -0,0 +1,101 @@
+package com.twitter.finagle.buoyant.h2
+package netty4
+
+import com.twitter.concurrent.AsyncQueue
+import com.twitter.finagle.stats.{StatsReceiver, NullStatsReceiver}
+import com.twitter.util.{Future, Promise, Return, Stopwatch, Throw}
+import io.netty.handler.codec.http2.{Http2StreamFrame, Http2HeadersFrame}
+
+/**
+ * TODO how should we go about resetting streams/transports?
+ */
+private[h2] class Netty4ClientStreamTransport(
+  val streamId: Int,
+  transport: H2Transport.Writer,
+  minAccumFrames: Int,
+  statsReceiver: StatsReceiver = NullStatsReceiver
+) {
+  require(minAccumFrames >= 2)
+
+  private[this] val recvq = new AsyncQueue[Http2StreamFrame]
+
+  @volatile private[this] var isRequestFinished, isResponseFinished = false
+  def isClosed = isRequestFinished && isResponseFinished
+
+  private[this] val closeP = new Promise[Unit]
+  def onClose: Future[Unit] = closeP
+
+  private[this] def setRequestFinished(): Unit = {
+    isRequestFinished = true
+    if (isClosed) { closeP.setDone(); () }
```

Comment:
yeah i ended up doing `; ()` in this branch because it ends up being a lot less visually heavy -- I can keep one-liners as one liners (whereas having `val _ = ..` feels too cumbersome for a one-liner) and it also throws off vertical alignment in a bunch of cases, e.g.:
```scala
foo.bar()
foo.bah()
val _ = foo.baz()
```
vs
```scala
foo.bar()
foo.bah()
foo.baz(); ()
```

I think the semicolon sends the same sort of _i'm doing something intentional here_ warning.  Not super attached to this, but I do feel it made some parts of this read quite a bit nicer.